### PR TITLE
Api controllers extension

### PR DESCRIPTION
**Pull request by skoppers**
Extended some functionallity of the controllers. Also fixed a lot of checkstyle errors.
The "VisibilityModifier" checkstyle module is now disabled for the model package, so we can use public variables without checkstyle throwing any errors.

### CODE REVIEW

File: app/models/Action.java [hunk 1]
```diff
@@ -35,13 +35,26 @@
...
+     * @param script    The script to which the action belongs.
+     * @return  The created Action object.
+     */
+    public static Action createAction(
+            String des, int timestamp, int duration, Preset preset, Script script) {
         Action act =  new Action();
         act.description = des;
         act.timestamp = timestamp;
-        act.estTime = estTime;
+        act.duration = duration;
```
Comment (hiddely) [inline app/models/Action.java:57]: Good change, this was bugging me..

File: app/controllers/Preset.java [hunk 2]
```diff
@@ -1,25 +1,54 @@
...
+/**
+ * Api controller for handling the json calls from the client.
+ */
 public class Preset extends Controller {
 
+    /**
+     * Get a list of all Preset objects in the database.
+     * @return The list of all Preset objects.
+     */
     public Result index() {
```
Comment (jorisoudejans) [inline app/controllers/Preset.java:20]: Maybe rename this to "getAll", like in Script?

File: checkstyle-config.xml [hunk 3]
```diff
@@ -116,4 +116,7 @@
   <module name="SuppressionCommentFilter"/>
   <module name="SuppressWithNearbyCommentFilter"/>
   <module name="SuppressWarningsFilter" />
+  <module name="SuppressionFilter">
```
Comment (jorisoudejans) [inline checkstyle-config.xml:119]: Awesome

File: suppressions.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<!--For the models package we would like to use public variables.
+    This because the classes are used as objects in a database,
+    so using public variables makes working with them easier.-->
+<suppressions>
```
Comment (jorisoudejans) [inline suppressions.xml:10]: Very awesome

File: app/controllers/Preset.java (hunk 2 above)
Comment (skoppers) [inline app/controllers/Preset.java:20]: Sure will do :+1:

**Pull request merged**